luxtronik/__init__.py

```diff
@@ -74,43 +88,50 @@ def __del__(self):
                 "Disconnected from Luxtronik heatpump %s:%s", self._host, self._port
             )
 
-    def read(self):
-        """Read data from heatpump."""
-        return self._read_after_write(parameters=None)
-
-    def write(self, parameters):
-        """Write parameter to heatpump."""
-        return self._read_after_write(parameters=parameters)
-
-    def _read_after_write(self, parameters):
+    def _with_lock_and_connect(self, func, *args, **kwargs):
         """
-        Read and/or write value from and/or to heatpump.
-        This method is essentially a wrapper for the _read() and _write()
-        methods.
+        Decorator around various read/write functions to connect first.
+
+        This method is essentially a wrapper for the _read() and _write() methods.
         Locking is being used to ensure that only a single socket operation is
         performed at any point in time. This helps to avoid issues with the
         Luxtronik controller, which seems unstable otherwise.
-        If write is true, all parameters will be written to the heat pump
-        prior to reading back in all data from the heat pump. If write is
-        false, no data will be written, but all available data will be read
-        from the heat pump.
+        """
+        with self._lock:
+            self._connect()
+            ret_val = func(*args, **kwargs)
+            # self._disconnect()
```

Comment:
This line can be deleted?

Reply:
Done. Btw.: It might be better to close the connection afterwards